### ISSUE TITLE
Add config option launch webserver command

### ISSUE THREAD

**Issue by janosh** [vitest-dev/vitest#1428]
### Clear and concise description of the problem

Playwright has the neat feature of launching a web server automatically before running tests:

https://playwright.dev/docs/test-advanced#launching-a-development-web-server-during-the-tests

```ts
// playwright.config.ts
import { PlaywrightTestConfig } from '@playwright/test';
const config: PlaywrightTestConfig = {
  webServer: {
    command: 'npm run start',
    url: 'http://localhost:3000/app/',
    timeout: 120 * 1000,
    reuseExistingServer: !process.env.CI,
  },
  use: {
    baseURL: 'http://localhost:3000/app/',
  },
};
export default config;
``` 

### Suggested solution

Would be great if `vitest` had this too.

### Additional context

Particularly useful for E2E tests which require the site being tested to be reachable.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by sheremet-va** [comment]
Can you give example how Vitest users can utilize it? Since we are more in a space of unit test frameworks, I don't see it in our scope.

**Comment by janosh** [comment]
Good question, I have a somewhat unusual setup in https://github.com/janosh/svelte-multiselect where I use `vitest` assertions to test the results of Playwright DOM manipulations. For example:

```ts
// tests/multiselect.test.ts
import * as playwright from 'playwright'
import { describe, expect, test } from 'vitest'

const context = await browser.newContext({
  baseURL: `http://localhost:${port}`,
})

test(`opens dropdown on focus`, async () => {
  const page = await context.newPage()
  await page.goto(`/ui`)

  expect(await page.$(`div.multiselect > ul.options.hidden`)).toBeTruthy()

  // ...

  const visibility = await page.$eval(
    `div.multiselect > ul.options`,
    (el) => getComputedStyle(el).visibility
  )
  expect(visibility).toBe(`visible`)
})
```

These tests only work if `http://localhost:${port}` is up and running which is a manual step atm. I could migrate my E2E tests fully to Playwright and use their `webServer` command but since I also have unit tests in `vitest` I'd prefer not to split assertions across different test runners.

**Comment by sheremet-va** [comment]
I agree that it's better to have a single library, but I don't think it is in scope of `vitest` 🤔 Since by default we don't even expect any server to be started. I would recommend the usual [start-server-and-test](https://github.com/bahmutov/start-server-and-test) package.

**Comment by janosh** [comment]
I've used [start-server-and-test](https://github.com/bahmutov/start-server-and-test) in the past but I like to keep a minimal list of dependencies.

I'll use this instead for now. Only downside is it crashes if trying to `npm run dev` after `npm run test` was already started.

```json5
// package.json
{
  "scripts": {
    "dev": "svelte-kit dev",
    "test": "npm run dev &\nvitest"
  }
}

**Comment by sheremet-va** [comment]
Out of scope of Vitest, unfortunately. Maybe it will be a part of https://github.com/vitest-dev/vitest/pull/1302.

**Comment by janosh** [comment]
Ah cool, nice to see #1302. But a bit confused. If it might be added down the road, we can leave this open?
